### PR TITLE
AZP: remove ws

### PR DESCRIPTION
**Pull request by avildema**
after cancel job next job cannot remove workspace

### CODE REVIEW

File: buildlib/azure-pipelines-pr.yml [hunk 1]
```diff
@@ -139,13 +137,28 @@ stages:
...
+              echo "dir: $(Build.Repository.LocalPath)"
+              lsof $(Build.Repository.LocalPath) || true
+              retry=3
+              while [[ ${retry} != 0 ]]
+              do
+                chmod 755 -R $(Build.Repository.LocalPath)
+                rm -rf $(Build.Repository.LocalPath) && break
+                sleep 5
+                retry=$((retry-1))
+              done
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:151]: maybe put this in az-helpers?
do we need it in other stages (e.g coverity, java, ..) ?
BTW, seems like it happens only when job running in docker

File: buildlib/azure-pipelines-pr.yml [hunk 2]
```diff
@@ -114,12 +114,10 @@ stages:
   - stage: Build
     jobs:
       - job: build_source
-        workspace:
-          clean: all
         pool:
           name: MLNX
           demands:
-            - ucx_docker -equals yes
+            - ucx_docker_test -equals yes
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:120]: why changed?

**Review by yosefe — COMMENTED** [review]
perf offline discussion, AFAIR we wanted to move build into docker FS, but this PR just seems to make all commands WORKSPACE-relative (which is good) but not changing workspace path, expect cov build dir. What am I missing?

File: buildlib/azure-pipelines-pr.yml [hunk 3]
```diff
@@ -165,26 +163,15 @@ stages:
 
           - bash: |
               set -eE
-              mkdir build && cd build
+              . buildlib/tools/common.sh
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:166]: ${buildlib}

File: buildlib/tools/common.sh [hunk 4]
```diff
@@ -1,7 +1,9 @@
 #!/bin/bash -eExl
 
 WORKSPACE=${WORKSPACE:=$PWD}
-ucx_inst=${WORKSPACE}/install
+# build in local directory which goes away when docker exits
+ucx_build_dir=$HOME/$BUILD_NUMBER/build
```
Comment (yosefe) [inline buildlib/tools/common.sh:5]: BUILD_NUMBER -> BUILD_ID

File: buildlib/tools/common.sh [hunk 5]
```diff
@@ -34,3 +36,22 @@ make_clean() {
...
+	cd ${WORKSPACE}
+	if [ -d ${ucx_build_dir} ]
+	then
+		chmod u+rwx ${ucx_build_dir} -R
+		rm -rf ${ucx_build_dir}
+	fi
+	./autogen.sh
+	mkdir -p ${ucx_build_dir}
+	cd ${ucx_build_dir}
+	export PROGRESS=0
```
Comment (yosefe) [inline buildlib/tools/common.sh:56]: what is this?

File: buildlib/azure-pipelines-pr.yml (hunk 3 above)
Comment (avildema) [reply buildlib/azure-pipelines-pr.yml:166]: ?
this is path

File: buildlib/tools/common.sh (hunk 5 above)
Comment (avildema) [reply buildlib/tools/common.sh:56]: for azure view, added progress bar for build process 

```sh
do_task() {
	$@
	echo "##vso[task.setprogress value=$PROGRESS;]Progress Indicator"
	PROGRESS=$((PROGRESS+5))
}
```

File: buildlib/azure-pipelines-pr.yml (hunk 3 above)
Comment (yosefe) [reply buildlib/azure-pipelines-pr.yml:166]: right

File: buildlib/tools/common.sh (hunk 5 above)
Comment (yosefe) [reply buildlib/tools/common.sh:56]: hah nice

**Review by yosefe — APPROVED** [review]

File: buildlib/tools/builds.sh [hunk 6]
```diff
@@ -247,6 +247,7 @@ build_clang() {
 		make_clean
 		$MAKEP
 		$MAKEP install
+		UCX_LOG_LEVEL_TRIGGER=ERROR $ucx_inst/bin/ucx_info -d
```
Comment (yosefe) [inline buildlib/tools/builds.sh:250]: export UCX_LOG_LEVEL_TRIGGER=error
export UCX_HANDLE_ERRORS=bt

File: buildlib/tools/builds.sh [hunk 7]
```diff
@@ -278,6 +279,7 @@ build_gcc() {
 			make_clean
 			$MAKEP
 			$MAKEP install
+			UCX_LOG_LEVEL_TRIGGER=ERROR $ucx_inst/bin/ucx_info -d
```
Comment (yosefe) [inline buildlib/tools/builds.sh:282]: maybe add helper `test_ucx_info <params>` function

File: buildlib/tools/builds.sh (hunk 6 above)
Comment (yosefe) [reply buildlib/tools/builds.sh:250]: remove

**Review by yosefe — APPROVED** [review]

**Pull request merged**